src/services/wishlistServices/addToWishlist.js

```diff
@@ -0,0 +1,16 @@
+import axios from "axios";
+
+const addToWishlist = async (token, product, setWishlist) => {
+  const response = await axios.post(
+    "/api/user/wishlist",
+    { product },
+    {
+      headers: {
+        authorization: token,
+      },
+    }
+  );
+  const wishlist = response.data.wishlist;
+  setWishlist(wishlist);
```

Comment:
Correct me if I am wrong but maybe you shouldn't do any state update in the service function.

Reply:
Thanks for the suggestion, will update it.